### PR TITLE
[Instrument] Prototype for Rendering Instruments in JSON Format

### PR DESCRIPTION
**Pull request by AnyhowStep**
This isn't meant to be pulled as-is. This is to just show a working demo and was built with information from `bmi` and `tsi`.

To test,

1. `checkout` this branch
1. Navigate to `<loris-host-name>/test_react_instrument`
1. Click "Render"

The page starts with `tsi` data but you should be able to put in any kind of valid instrument JSON and run it. I'll need more instrument data before I can build more of this.

image

WIP
image

Multi-page
image

### CODE REVIEW

**Comment by AnyhowStep** [comment]
http://loris-17-1-issue-tracker.anyhowstep.com/test_react_instrument/

File: modules/test_react_instrument/jsx/InputDate.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import InputBase from "./InputBase";
+
```
Comment (alisterdev) [inline modules/test_react_instrument/jsx/InputDate.js]: I feel like you are trying to rewrite all the form elements in React, when such thing was already done [here](https://github.com/aces/Loris/blob/17.1-dev/jsx/Form.js)

Moreover, `<FormElement/>` accepts PHP Loris form representation as `formElements` prop. which usually is referred as`$this->form->form` in PHP code.

File: modules/test_react_instrument/jsx/InputDate.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+export default class InputDate extends InputBase {
```
Comment (jacobpenny) [inline modules/test_react_instrument/jsx/InputDate.js]: Composition is strongly preferred over inheritance in React:
https://facebook.github.io/react/docs/composition-vs-inheritance.html
http://stackoverflow.com/a/36994501/2070413

**Comment by ridz1208** [comment]
@jstirling91 @driusan @AnyhowStep 

As this is simply a proof of concept which was made before workgroups existed, this PR can either be closed or just left on "blocked" for the moment

**Comment by driusan** [comment]
@jstirling91 you're leading the instrument working group, I'll leave the call on @ridz1208's point to you if you want to close this or not..

**Comment by jstirling91** [comment]
@davidblader @jacobpenny what's you thoughts on this. Is this a good starting point for reactifing instruments?

**Comment by jacobpenny** [comment]
@jstirling91 It goes against two practices that I would like to enforce in our React code. One, which I mentioned above, is that it uses inheritance instead of composition. The second is that it doesn't use [controlled components](https://facebook.github.io/react/docs/forms.html#controlled-components) (like those found in `Form.js`). I appreciate the work done here (and I'm sure we can leverage some of it) but I'm in favour of a more idiomatic implementation.

**Comment by jstirling91** [comment]
In that case I'm closing the pull request

**Pull request closed without merging**